src/rgw/rgw_rest.cc

```diff
@@ -382,11 +382,22 @@ void dump_header(struct req_state* const s,
   }
 }
 
+static inline boost::string_ref get_sanitized_hdrval(ceph::buffer::list& raw)
+{
+  /* std::string and thus boost::string_ref ARE OBLIGED to carry multiple
+   * 0x00 and count them to the length of a string. We need to take that
+   * into consideration and sanitize the size of a ceph::buffer::list used
+   * to store metadata values (x-amz-meta-*, X-Container-Meta-*, etags).
+   * Otherwise we might send 0x00 to clients. */
+  const size_t len = raw.length();
+  return boost::string_ref(raw.c_str(), len ? len - 1 : 0);
```

Comment:
> i'm not sure why we need a null terminator in these bufferlists, because we shouldn't be trying to print them out

I guess it was a workaround for `ceph::buffer::list::c_str()` that AFAIK doesn't guarantee its caller to obtain a *null-terminated* C-string.